File: leetcode/python/2.两数相加.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        head = ListNode(0)
        start = head
        c = 0
        while c or l1 or l2:
            if l1:
                c += l1.val
                l1 = l1.next
            if l2:
                c += l2.val
                l2 = l2.next
            
            val, c = c % 10, c // 10
            start.next = ListNode(val)
            
            start = start.next
        
        return head.next

File: leetcode/golang/2.两数相加.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    c := 0;
    var head ListNode;
    start := &head;

    for {
        if c == 0 && l1 == nil && l2 == nil {
            break;
        }

        if l1 != nil {
            c += l1.Val;
            l1 = l1.Next;
        }
        if l2 != nil {
            c += l2.Val;
            l2 = l2.Next;
        }

        val := c % 10;
        c = c / 10;

        var tmp ListNode;
        tmp.Val = val;

        start.Next = &tmp;
        start = &tmp;
    }
    return head.Next;
}